tests/ops/service/connectors/test_saas_connector.py

```diff
@@ -423,3 +426,192 @@ def test_get_consent_requests_by_preference(
+
+    def test_missing_identity_data_failure(
+        self,
+        db,
+        system,
+        consent_policy,
+        mailchimp_transactional_connection_config_no_secrets,
+        privacy_preference_history,
+    ):
+        """We need a matching identity for the connector in order to send the request
+        Mailchimp Transactional set up to fail if no email supplied
+        """
+        mailchimp_transactional_connection_config_no_secrets.system_id = system.id
+        privacy_request = PrivacyRequest(
+            id=f"test_consent_request_task_{random.randint(0, 1000)}",
+            status=PrivacyRequestStatus.pending,
+        )
+        privacy_request.save(db)
+        privacy_preference_history.privacy_request_id = privacy_request.id
+        privacy_preference_history.save(db)
+
+        connector = get_connector(mailchimp_transactional_connection_config_no_secrets)
+        with pytest.raises(ValueError):
+            connector.run_consent_request(
+                node=TraversalNode(generate_node("a", "b", "c", "c2")),
+                policy=consent_policy,
+                privacy_request=privacy_request,
+                identity_data={"ljt_readerID": "abcde"},
+            )
+
+    def test_missing_identity_data_skipped(
+        self,
+        db,
+        system,
+        consent_policy,
+        google_analytics_connection_config_without_secrets,
+        privacy_preference_history,
+    ):
+        """We need a matching identity for the connector in order to send the google_analytics_connection_config_without_secrets
+        Google Analytics set up to skip instead of fail if we don't have the ga client id.
+        There's no guarantee that a ga cookie is in the browser
+        """
+        google_analytics_connection_config_without_secrets.system_id = system.id
+
+        privacy_request = PrivacyRequest(
+            id=f"test_consent_request_task_{random.randint(0, 1000)}",
+            status=PrivacyRequestStatus.pending,
+        )
+        privacy_request.save(db)
+        privacy_preference_history.privacy_request_id = privacy_request.id
+        privacy_preference_history.save(db)
+
+        connector = get_connector(google_analytics_connection_config_without_secrets)
+        with pytest.raises(SkippingConsentPropagation) as exc:
+            connector.run_consent_request(
+                node=TraversalNode(generate_node("a", "b", "c", "c2")),
+                policy=consent_policy,
+                privacy_request=privacy_request,
+                identity_data={"ljt_readerID": "abcde"},
+            )
+
+        assert "Missing needed values to propagate request" in str(exc)
+
+    def test_no_requests_of_that_type_defined(
+        self,
+        system,
+        consent_policy,
+        google_analytics_connection_config_without_secrets,
+        privacy_preference_history,
+        db,
+    ):
+        """User is expressing an opt in preference here but GA only has an opt out preference defined"""
```

Comment:
sure thing! this behavior isn't new by the way and is end-to-end tested in the google analytics connector but I thought it would be good to add some more intermediate tests to this method while I'm here (also codecov likely flagged this if I recall)

Reply:
oh yeah, i just meant all the docstrings in your test methods generally :)

Reply:
oh good!